src/clj/rems/api/applications_v2.clj

```diff
@@ -0,0 +1,810 @@
+                            :field/max-length (:maxlength item)})
+                         (:items form))
+        fields (merge-lists-by :field/id rich-fields app-fields)
+        description (->> fields
+                         (filter #(= :description (:field/type %)))
+                         first
+                         :field/value)]
+    (-> application
+        (assoc-in [:application/form :form/title] (:title form))
+        (assoc-in [:application/form :form/fields] fields)
+        (assoc :application/description description))))
+
+(defn- enrich-resources [app-resources get-catalogue-item]
+  (->> app-resources
+       (map :catalogue-item/id)
+       (map get-catalogue-item)
+       (map (fn [item]
+              {:catalogue-item/id (:id item)
+               :resource/id (:resource-id item)
+               :resource/ext-id (:resid item)
+               :catalogue-item/title (assoc (localization-for :title item)
+                                            :default (:title item))
+               :catalogue-item/start (:start item)
+               :catalogue-item/state (keyword (:state item))}))
+       (sort-by :catalogue-item/id)))
+
+(defn- enrich-licenses [app-licenses get-license]
+  (let [rich-licenses (->> app-licenses
+                           (map :license/id)
+                           (map get-license)
+                           (map (fn [license]
+                                  (let [type (keyword (:licensetype license))
+                                        content-key (case type
+                                                      :link :license/link
+                                                      :text :license/text)]
+                                    {:license/id (:id license)
+                                     :license/type type
+                                     :license/start (:start license)
+                                     :license/end (:end license)
+                                     :license/title (assoc (localization-for :title license)
+                                                           :default (:title license))
+                                     content-key (assoc (localization-for :textcontent license)
+                                                        :default (:textcontent license))})))
+                           (sort-by :license/id))]
+    (merge-lists-by :license/id rich-licenses app-licenses)))
+
+(defn- assoc-injections [application {:keys [get-form get-catalogue-item get-license get-user]}]
+  (-> application
+      (enrich-form get-form)
+      (update :application/resources enrich-resources get-catalogue-item)
+      (update :application/licenses enrich-licenses get-license)
+      (assoc :application/applicant-attributes (get-user (:application/applicant application)))))
+
+(defn- build-application-view [events injections]
+  (-> (reduce application-view nil events)
+      (assoc-injections injections)))
```

Comment:
Maybe @opqdonut can comment as well but I don't think this `assoc-injections` name clarifies at all what is happening here but only hides details. Do you like this pattern? I'd name this like `join-externals` or just have each of the "injections" lines of code here by themselves or name each one like `join-resources` or `enrich-resources`...

Reply:
I think it's fine. The name could be `enrich-with-injections` or something. These are private functions and grouping them a bit instead of having one huge `->` is nice.

Reply:
I'd prefer inlining that function and the "huge" `->` is exactly what I prefer here where each line in the expression is "at the same level of abstraction" and the same "shape".

Reply:
Renamed to `enrich-with-injections`. Inlining it is not an option, because it is called from multiple places: the listing for all applications (`get-user-applications-v2`) needs to do the same things as `build-application-view` but in a bit different order.